src/main/java/data_access/StockDataAccessObject.java

```diff
@@ -6,103 +6,157 @@
                 Request profileRequest = new Request.Builder().url(profileUrl).build();
 
-                // Quote api call to get current market executionPrice
-                try (Response quoteResponse = client.newCall(quoteRequest).execute()) {
-                    if (quoteResponse.isSuccessful()) {
-                        String quoteResponseBody = quoteResponse.body().string();
-                        JSONObject jsonObject = new JSONObject(quoteResponseBody);
-                        price = jsonObject.getDouble("c");
-                    } else {
-                        System.out.println("API call limit exceeded.");
-                    }
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-
-                // Profile2 api call to get company name and industry
+                // Initiates API call request
                 try (Response profileResponse = client.newCall(profileRequest).execute()) {
                     if (profileResponse.isSuccessful()) {
-                        String profileResponseBody = profileResponse.body().string();
-                        JSONObject jsonObject = new JSONObject(profileResponseBody);
-                        company = jsonObject.getString("name");
-                        industry = jsonObject.getString("finnhubIndustry");
+                        JSONObject jsonObject = new JSONObject(profileResponse.body().string());
+                        // Gets the ticker company "name" or returns default company name if unavailable
+                        company = jsonObject.optString("name", company);
+                        // Gets the company's industry "finnhubIndustry" based on finnhub's classification or returns default industry if unavailable
+                        industry = jsonObject.optString("finnhubIndustry", industry);
                     } else {
-                        System.out.println("API call limit exceeded.");
+                        System.out.println("Failed to fetch profile data for ticker: " + ticker);
                     }
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
                 }
 
-                // Creates new Stock with retrieved information
-                Stock stock = new Stock(ticker, company, industry, price);
-                stocks.put(line, stock);
+                // Check if the stock already exists in the map
+                if (stocks.containsKey(ticker)) {
+                    // Update the price of the existing stock
+                    stocks.get(ticker).updatePrice(price);
+                } else {
+                    // Create a new Stock object and add it to the map
+                    Stock stock = new Stock(ticker, company, industry, price);
+                    stocks.put(ticker, stock);
+                }
+
+            } catch (Exception e) {
```

Comment:
Here we cannot assume that any type of exception is caused by exceeding rate limit. Is there a way to determine that a rate limit occurs (such as status code from the API call)? Throw `RateLimitExceededException` only when the error is indeed caused by exceeding the limit, otherwise we can simply call `printStackTrace` to display the error, though we will need to switch to a more robust alternative in the future.

Reply:
Status code 429 indicates rate limit exceeded, and others should be common http status codes.